apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -1941,6 +1969,377 @@ export class LemonadePipeline implements BasePipeline {
 
     return ids;
   }
+
+  /**
+   * Returns {@link MemberCriteria} for each of the ticket types this user has
+   * permission to check in.
+   */
+  private async ticketsUserCanCheckIn(
+    checkerEmail: string
+  ): Promise<MemberCriteria[]> {
+    if (this.definition.options.superuserEmails?.includes(checkerEmail)) {
+      // The user can check in all of the event/product combinations on this
+      // pipeline
+      return this.definition.options.events.flatMap((eventConfig) =>
+        eventConfig.ticketTypes.map((ticketType) => ({
+          eventId: eventConfig.genericIssuanceEventId,
+          productId: ticketType.genericIssuanceProductId
+        }))
+      );
+    }
+
+    // Get all of the products that the checker owns
+    const checkerProductIds: string[] = [];
+    for (const checkerTicketAtom of await this.db.loadByEmail(
+      this.id,
+      checkerEmail
+    )) {
+      checkerProductIds.push(checkerTicketAtom.productId);
+    }
+    for (const manualTicket of this.getManualTicketsForEmail(checkerEmail)) {
+      checkerProductIds.push(manualTicket.productId);
+    }
+
+    // Return any event/product combinations for which the user has a superuser
+    // ticket.
+    return this.definition.options.events.reduce((memo, eventConfig) => {
+      if (
+        // Does the user own a superuser product for this event?
+        eventConfig.ticketTypes.some(
+          (ticketType) =>
+            ticketType.isSuperUser &&
+            checkerProductIds.includes(ticketType.genericIssuanceProductId)
+        )
+      ) {
+        // In that case, they can check in any of the product types for this event
+        memo.push(
+          ...eventConfig.ticketTypes.map((ticketType) => ({
+            eventId: eventConfig.genericIssuanceEventId,
+            productId: ticketType.genericIssuanceProductId
+          }))
+        );
```

Comment:
in `MemberCriteria`, the `productId` field is optional, and in the case that it's absent, it represents the fact that any product type in the given event as identified by the `eventId` matches the `MemberCriteria`. Thus, I believe this could be simplified by just adding a single member criteria here, where the event id is `eventConfig.genericIssuanceEventId` and the `productId` is `undefined`